tests/unit/s2n_connection_serialize_test.c

```diff
@@ -473,6 +490,271 @@ int main(int argc, char **argv)
         EXPECT_OK(s2n_send_and_recv_test(client_conn, new_server_conn));
     };
 
+    /* Self-talk: Test interaction between resumption and TLS transfer. */
+    if (s2n_is_tls13_fully_supported()) {
+        DEFER_CLEANUP(struct s2n_config *resumption_config = s2n_config_new(), s2n_config_ptr_free);
+        EXPECT_SUCCESS(s2n_config_add_cert_chain_and_key_to_store(resumption_config, chain_and_key));
+        EXPECT_SUCCESS(s2n_config_disable_x509_verification(resumption_config));
+        EXPECT_SUCCESS(s2n_config_set_serialized_connection_version(resumption_config, S2N_SERIALIZED_CONN_V1));
+        EXPECT_SUCCESS(s2n_config_set_cipher_preferences(resumption_config, "default_tls13"));
+        EXPECT_SUCCESS(s2n_config_set_session_tickets_onoff(resumption_config, true));
+
+        /* Client resumption configuration */
+        DEFER_CLEANUP(struct s2n_stuffer cb_session_data = { 0 }, s2n_stuffer_free);
+        EXPECT_SUCCESS(s2n_stuffer_growable_alloc(&cb_session_data, 0));
+        EXPECT_SUCCESS(s2n_config_set_session_ticket_cb(resumption_config, s2n_test_session_ticket_cb, &cb_session_data));
+
+        /* Server resumption configuration */
+        S2N_BLOB_FROM_HEX(ticket_key,
+                "077709362c2e32df0ddc3f0dc47bba63"
+                "90b6c73bb50f9c3122ec844ad7c2b3e5");
+        uint64_t current_time = 0;
+        uint8_t ticket_key_name[16] = "resumption key\0";
+        EXPECT_SUCCESS(resumption_config->wall_clock(resumption_config->sys_clock_ctx, &current_time));
+        EXPECT_SUCCESS(s2n_config_add_ticket_crypto_key(resumption_config, ticket_key_name, strlen((char *) ticket_key_name),
```

Comment:
This is some pretty messy setup. I'm surprised there isn't a testlib method for it.